src/webportal/src/app/home/index.jsx

```diff
@@ -56,6 +56,10 @@ if (config.authnMethod === 'OIDC') {
   }
 }
 
+if (query.has('loginFailMsg')) {
```

Comment:
should check the login failure right after line 41 and redirect to index page if login failed

Reply:
I think return to the source page not index page will be better.